### PR TITLE
[MRG+1] Release 1.2.0

### PR DESCRIPTION
**Pull request by tgsmith61591**
### v1.2.0

Version 1.2.0 is a feature release that also patches several known bugs. Notable changes include:

* Adds the `OCSBTest` of seasonality, as discussed in #88
* Default value of `seasonal_test` changes from “ch” to “ocsb” in `auto_arima`
* Default value of `test` changes from “ch” to “ocsb” in nsdiffs
* Adds benchmarking notebook and capabilities in pytest plugins
* Removes the following environment variables, which are now deprecated:
    - `PMDARIMA_CACHE` and `PYRAMID_ARIMA_CACHE`
    - `PMDARIMA_CACHE_WARN_SIZE` and `PYRAMID_ARIMA_CACHE_WARN_SIZE`
    - `PYRAMID_MPL_DEBUG`
    - `PYRAMID_MPL_BACKEND`
* Deprecates the `is_stationary` method in tests of stationarity. This will be removed in v1.4.0. Use `should_diff` instead.
* Adds two new datasets: `airpassengers` & `austres`
* When using `out_of_sample`, the out-of-sample predictions are now stored under the `oob_preds_` attribute.
* Adds a number of transformer classes including:
    - `BoxCoxEndogTransformer`
    - `FourierFeaturizer`
* Adds a `Pipeline` class resembling that of scikit-learn’s, which allows the stacking of transformers together.
* Adds a class wrapper for `auto_arima`: `AutoARIMA`. This is allows auto-ARIMA to be used with pipelines.

### CODE REVIEW

**Comment by tgsmith61591** [comment]
To be released 4/27. Weigh in here with final thoughts...

**Comment by aaronreidsmith** [comment]
Don't forget to drop the `dev` tag from the [version](https://github.com/tgsmith61591/pmdarima/blob/20b50ec7dfa681554b4c4655c52e86a6a2929132/pmdarima/__init__.py#L9)

File: pmdarima/__init__.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import os as _os
 
-__version__ = "1.1.1"
+__version__ = "1.2.0-dev1"
```
Comment (aaronreidsmith) [inline pmdarima/__init__.py]: Should have just left my comment here 🙄

**Comment by aaronreidsmith** [comment]
Do we want to add Python 3.7 to our [PyPI classifiers](https://github.com/tgsmith61591/pmdarima/blob/86824fac0f6cdb25a33b55ccc5d9da4d612a5ab7/setup.py#L193)? Probably should have added that in `1.1.0`...

**Comment by aaronreidsmith** [comment]
Everything LGTM! 🎉

**Pull request merged**